packages/gradle/src/plugin/utils/create-ci-targets.ts

```diff
@@ -0,0 +1,148 @@
+import { TargetConfiguration } from '@nx/devkit';
+import { GradlePluginOptions } from './gradle-plugin-options';
+import { isCI } from 'nx/src/devkit-internals';
+import { RunCommandsOptions } from 'nx/src/executors/run-commands/run-commands.impl';
+
+/**
+ * Replace target name with options
+ * if ciTargetName is provided, replace target name "ci" with ciTargetName
+ * if isCI, replace target test in dependsOn with ci and exclude test in each command
+ */
+export function replaceTargetNameWithOptions(
```

Comment:
not saying it's wrong but is there a reason we replace the ciTargetName after the fact instead of passing ciTargetName into the gradle task? That way it could set up the correct naming right away

Reply:
good point, move this logics to plugin